identity-iota/src/tangle/message_version.rs

```diff
@@ -0,0 +1,41 @@
+// Copyright 2020-2021 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::Error;
+
+#[derive(Copy, Clone)]
+pub(crate) enum MessageVersion {
+  V1 = 1,
+}
+
+impl MessageVersion {
+  pub(crate) const CURRENT: Self = MessageVersion::V1;
+
+  /// Adds the current message version flag at the beginning of arbitrary data.
+  pub(crate) fn add_version_flag(mut data: Vec<u8>, message_version: MessageVersion) -> Vec<u8> {
+    let version_flag = message_version as u8;
+    data.splice(0..0, [version_flag].iter().cloned());
+    data
+  }
+
+  /// Checks if flag matches message_version.
+  pub(crate) fn check_version_flag(flag: &u8, message_version: MessageVersion) -> Result<(), Error> {
+    if message_version as u8 == *flag {
+      Ok(())
+    } else {
+      Err(Error::InvalidMessageFlags)
+    }
+  }
+}
+
```

Comment:
It feels very adhoc and non-idiomatic Rust to have these as associated functions. I would suggest to either (1): turn them into methods so one would instead have: 
```rust
fn add version_flag(&self, mut data: Vec<u8>) -> Vec<u8>

fn check_version_flag(&self, flag: &u8) -> Result<(), Error>
```
or alternatively one can (2): rename the module to `message_versioning`
and move `add_version_flag` and `check_version_flag` out of `MessageVersion` and then one would call them 
with `message_versioning::add_version_flag`, `message_versioning::check_version_flag` elsewhere in the crate. 

What do you think @cycraig ?